cabal-install/Distribution/Solver/Modular/Tree.hs

```diff
@@ -37,17 +37,17 @@ type Weight = Double
 -- package, flag, and stanza choices control the traversal order.
 -- TODO: The weight type should be changed from [Double] to Double to avoid
 -- giving too much weight to preferences that are applied later.
-data Tree a =
+data Tree a b =
```

Comment:
I added comments describing the type variables for `Tree` and `TreeF`.  I also renamed the types `c` and `d`, for choice and done nodes, respectively.  I think that the one-letter names are more readable when they appear several times in a single type signature.